src/Sulu/Bundle/AdminBundle/Resources/js/components/Popover/PopoverPositioner.js

```diff
@@ -0,0 +1,112 @@
+// @flow
+import type {PopoverDimensions, PopoverStyle, VerticalCrop} from './types';
+
+const MIN_HEIGHT = 100;
+const PADDING_TO_WINDOW = 10;
+
+/**
+ * The class is responsible for calculating the position of the popover on the screen when opened.
+ * Generally, it positions the popover above an anchor element. Moreover the popover is shifted, such that
+ * a designated element in the popover is right above the anchor element.
+ * It is made sure that the popover does not overflow the borders of the screen and even keeps a padding to the borders.
+ *
+ * The behaviour described to this point, however, is broken at one point: When the height of the popover undercuts
+ * a certain minimum height, the popover is positioned from the anchor element upwards or downwards depending on if
+ * the popover overflows the bottom or top border of the screen.
+ */
+export default class PopoverPositioner {
+    static dimensionsToStyle(dimensions: PopoverDimensions): PopoverStyle {
+        return {
+            top: dimensions.top + 'px',
+            height: dimensions.height + 'px',
+            left: dimensions.left + 'px',
+        };
+    }
+
+    static getCroppedDimensions(
+        popoverWidth: number,
+        popoverHeight: number,
+        anchorTop: number,
+        anchorLeft: number,
+        anchorWidth: number,
+        anchorHeight: number,
+        horizontalOffset: number,
+        verticalOffset: number,
+        centerChildOffsetTop: number,
+        alignOnVerticalAnchorEdges: boolean = true,
+        windowWidth: number = window.innerWidth,
+        windowHeight: number = window.innerHeight,
```

Comment:
Are the two window properties just here for testing? Then I would remove them, and set them in the test using `window.innerWidth` resp. `window.innerHeight`.